### DISCUSSION TITLE
Value requested for nonexistent variable

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#28309] [Q&A Modules: General]
Hi all, 

I am runing a coupled simulation in serial manner (not parallel). I need to call two outputs from previous steps to initialise my current model. I applyed "SolutionUserObject" to reed the data from the each outputs (I did not use "restart" because i needed to have the data in the functions). here is the blocks in input file: 

```
[UserObjects]
  [./solution_1] 
    type = SolutionUserObject
    mesh = output1.e
    timestep = LATEST
    system_variables = 'sigma_xx sigma_xy sigma_xz sigma_yy sigma_yz sigma_zz'
  [../]
     [./solution_2] 
    type = SolutionUserObject
    mesh = output2.e
    timestep = LATEST
    system_variables = 'stress_xx stress_xy stress_xz stress_yx stress_yy stress_yz stress_zx stress_zy stress_zz pressure temperature'
  [../]
[]
```

But the run terminates with the following error, while  'stress_zz' is in the list of available variables, it is shown as nonexistent variable.

```
 Currently Executing
  Performing Initial Setup................................................

*** ERROR ***
/export/dump/rahim/init_seissol-first.i:47.6:
The following error occurred in the UserObject 'solution_2' of type SolutionUserObject.

Value requested for nonexistent variable ' stress_zz' in the 'solution_2' SolutionUserObject.
System selected: nl0
Available variables:
 stress_xx, stress_xy, stress_xz, stress_yx, stress_yy, stress_yz, stress_zx, stress_zy, stress_zz, pressure, temperature


application called MPI_Abort(MPI_COMM_WORLD, 1) - process 42
```

Thanks, 
Rahim

**Comment by GiudGiud** [comment]
Hello

I m surprised the stresses are all in the 'nl0' (nonlinear) system.
Should you be pulling from the auxiliary system instead?

**Reply by GiudGiud** [reply]
Also I think the problem does not come from the solution user object, but from the SolutionIC or SolutionAux which instead of specifying 'stress_zz' specified ' stress_zz'

**Reply by Rahim-Habibi** [reply]
Thanks @GiudGiud , 

I have tried 'stress_zz' instead of ' stress_zz' and it worked. 
Rahim
